server/src/main/java/org/opensearch/action/admin/indices/mapping/get/GetFieldMappingsResponse.java

```diff
@@ -116,6 +117,10 @@ public class GetFieldMappingsResponse extends ActionResponse implements ToXConte
             String index = in.readString();
             if (in.getVersion().before(Version.V_2_0_0)) {
                 int typesSize = in.readVInt();
+                if (typesSize == 0) {
```

Comment:
@gaobinlong could you please why do we need that and when ``typesSize` could be 0?

In 2.x, we do:
```
            if (out.getVersion().before(Version.V_2_0_0)) {
                out.writeVInt(1);
                out.writeString(MapperService.SINGLE_MAPPING_NAME);
            }
```

For 1.x, types is never empty (afaik). Also, the `indexMapBuilder` is always populated on line 134, no matter of  `typesSize`

Reply:
In 1.x, when the requested field doesn't exist, then [typeMappings](https://github.com/opensearch-project/OpenSearch/blob/6447300608f9cff98e16ee8570651a0149c72520/server/src/main/java/org/opensearch/action/admin/indices/mapping/get/TransportGetFieldMappingsIndexAction.java#L144) is an empty map, resulting in [type size is 0](https://github.com/opensearch-project/OpenSearch/blob/6447300608f9cff98e16ee8570651a0149c72520/server/src/main/java/org/opensearch/action/admin/indices/mapping/get/GetFieldMappingsResponse.java#L306) when serializing the `GetFieldMappingsResponse` object.

Reply:
1.x node can deserialize response from 2.x node well, but 2.x node cannot deserialize response from 1.x node because the type size could be `0`.

Reply:
In this case `404` should be correct response for both version, right?  - the requested field doesn't exist

Reply:
No, 200 status code with empty response like `{ "test": {"mappings":{}} }` is as expected, see this yml test: https://github.com/opensearch-project/OpenSearch/blob/6447300608f9cff98e16ee8570651a0149c72520/rest-api-spec/src/main/resources/rest-api-spec/test/indices.get_field_mapping/20_missing_field.yml#L21.

Reply:
Makes sense, thank you @gaobinlong , could you please add the comment to the `if` block (https://github.com/opensearch-project/OpenSearch/pull/13624/files#r1596770728), thank you

Reply:
Done it, thank you for the reminding.